### PR TITLE
chore: shared staking queries

### PR DESCRIPTION
**Pull request by troykessler**
## Summary by CodeRabbit

- **New Features**
	- Enhanced querying capabilities for stakers with updated methods focusing on total stake rather than delegation.
	- Introduced new test cases for invalid and valid bundle scenarios, improving coverage for edge cases.
	- Added functionality for pagination and sorting of stakers based on pool stake.
	- Updated data structures and message formats to improve representation of staker information and pool stakes.

- **Bug Fixes**
	- Adjusted assertions in tests to align with new logic for total stake calculations and voting power constraints.

- **Documentation**
	- Updated comments and documentation to reflect changes in staker status and querying methods.

- **Tests**
	- Expanded test cases to cover new scenarios and refined existing tests for better accuracy in stake management.

- **Chores**
	- Introduced a new method to streamline setting maximum voting power in tests.

### CODE REVIEW

File: x/query/keeper/grpc_query_stakers_by_pool.go [hunk 1]
```diff
@@ -15,24 +16,22 @@ func (k Keeper) StakersByPool(c context.Context, req *types.QueryStakersByPoolRe
...
-				Staker:     k.GetFullStaker(ctx, valaccount.Staker),
-				Valaccount: valaccount,
-			})
-		}
+		stakers = append(stakers, *k.GetFullStaker(ctx, valaccount.Staker))
 	}
 
-	return &types.QueryStakersByPoolResponse{Stakers: data}, nil
+	sort.Slice(stakers, func(i, j int) bool {
+		return k.stakerKeeper.GetValidatorPoolStake(ctx, stakers[i].Address, req.PoolId) > k.stakerKeeper.GetValidatorPoolStake(ctx, stakers[j].Address, req.PoolId)
```
Comment (mbreithecker) [inline x/query/keeper/grpc_query_stakers_by_pool.go]: Maybe cache the values?

File: x/stakers/keeper/exported_functions.go [hunk 2]
```diff
@@ -193,7 +262,7 @@ func (k Keeper) GetValidatorPoolStakes(ctx sdk.Context, poolId uint64, mustInclu
 	totalStakeRemainder := totalStake
 
 	// sort descending based on stake
-	sort.SliceStable(validators, func(i, j int) bool {
+	sort.Slice(validators, func(i, j int) bool {
```
Comment (mbreithecker) [inline x/stakers/keeper/exported_functions.go]: Not in sync with main

**Review by mbreithecker — APPROVED** [review]
